lib/kitchen/directions/move_solutions_to_answer_key/strategies/precalculus.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Kitchen::Directions::MoveSolutionsToAnswerKey
+  module Strategies
+    class Precalculus
+      def bake(chapter:, append_to:)
+        bake_try_note(chapter: chapter, append_to: append_to)
+
+        # Bake section exercises
+        chapter.non_introduction_pages.each do |page|
+          number = "#{chapter.count_in(:book)}.#{page.count_in(:chapter)}"
+          bake_section(chapter: page, append_to: append_to, klass: 'section-exercises',
+                       number: number)
+        end
+
+        # Bake other types of exercises
+        classes = %w[review-exercises practice-test]
+        classes.each do |klass|
+          bake_section(chapter: chapter, append_to: append_to, klass: klass)
+        end
+      end
+
+      protected
+
+      def bake_section(chapter:, append_to:, klass:, number: nil)
+        section_solutions_set = []
+        chapter.search(".#{klass}").each do |section|
+          section.search('[data-type="solution"]').each do |solution|
+            section_solutions_set.push(solution.cut)
+          end
+        end
+
+        return if section_solutions_set.empty?
+
+        title = <<~HTML
+          <h3 data-type="title">
+            <span class="os-title-label">#{I18n.t(:"eoc.#{klass}", number: number)}</span>
+          </h3>
+        HTML
+
+        append_solution_area(title: title, solutions: section_solutions_set, append_to: append_to)
+      end
+
+      def bake_try_note(chapter:, append_to:)
```

Comment:
let's move this into another direction, seams cleaner than to have it in here

Reply:
discussed on call -- I'll rename this for clarity